src/vulkan/vertex_buffer.cc

```diff
@@ -23,70 +23,78 @@
 
 namespace amber {
 namespace vulkan {
+namespace {
+
+VkVertexInputRate GetVkInputRate(InputRate rate) {
+  if (rate == InputRate::kVertex)
+    return VK_VERTEX_INPUT_RATE_VERTEX;
+  else
+    return VK_VERTEX_INPUT_RATE_INSTANCE;
+}
+
+}  // namespace
 
 VertexBuffer::VertexBuffer(Device* device) : device_(device) {}
 
 VertexBuffer::~VertexBuffer() = default;
 
-void VertexBuffer::SetData(uint8_t location, Buffer* buffer) {
+void VertexBuffer::SetData(uint8_t location, Buffer* buffer, InputRate rate) {
   auto format = buffer->GetFormat();
-
+  const uint32_t binding = static_cast<uint32_t>(vertex_attr_desc_.size());
   vertex_attr_desc_.emplace_back();
-  // TODO(jaebaek): Support multiple binding
-  vertex_attr_desc_.back().binding = 0;
+  vertex_attr_desc_.back().binding = binding;
   vertex_attr_desc_.back().location = location;
-  vertex_attr_desc_.back().offset = stride_in_bytes_;
+  vertex_attr_desc_.back().offset = 0u;
   vertex_attr_desc_.back().format = device_->GetVkFormat(*format);
 
-  stride_in_bytes_ += format->SizeInBytes();
-  data_.push_back(buffer);
-}
-
-Result VertexBuffer::FillVertexBufferWithData(CommandBuffer* command) {
```

Comment:
So, why can this just go away? If we were interleaving buffers previously, doesn't that mean there are tests out there which many require this functionality?

Reply:
The interleaving was all happening under the hood (in Vulkan engine) and there was no benefit of doing that. Using separate buffers works just as fine. And this is not visible to shaders either. And as a bonus the code got a bit more simple.